client/src/pages/Balance.js

```diff
@@ -0,0 +1,100 @@
+import React, { useContext, useState } from "react";
+import userContext from "../userContext";
+import Background from "../elements/Background";
+import StyledPaper from "../elements/StyledPaper";
+import PageHeader from "../elements/PageHeader";
+import {
+  Button,
+  Card,
+  Container,
+  CssBaseline,
+  Divider,
+  Grid,
+  Typography,
+} from "@material-ui/core";
+
+const BalancePage = ({ state, dispatch }) => {
+  const user = useContext(userContext);
+  const [topUp, setTopUp] = useState(2);
+
+  const handleClick = ({ target, ...e }) => {
+    // e.prototype.preventDefault();
+    const { id } = target.closest("button");
```

Comment:
any reason not to do e.target.id?

Reply:
The `Button` component creates a button element that wraps some spans, which means the event target is actually the span that acts as the button label. 
```html
<button class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-sizeSmall" 
  tabindex="0" type="button" id="decr"
>
  <span class="MuiButton-label">-</span>
  <span class="MuiTouchRipple-root"></span>
</button>
```

Reply:
It looks like the `.MuiTouchRipple-root` has some click and focus handlers that spawn additional child nodes